src/__tests__/extensions/surveys.test.ts

```diff
@@ -31,28 +40,324 @@ describe('survey display logic', () => {
+        surveyDiv.attachShadow({ mode: 'open' })
+        surveyDiv.shadowRoot!.appendChild(document.createElement('style'))
+        document.body.appendChild(surveyDiv)
+
+        expect(surveyManager.getTestAPI().canShowNextEventBasedSurvey()).toBe(true)
+
+        surveyDiv.shadowRoot!.appendChild(document.createElement('div'))
+        expect(surveyManager.getTestAPI().canShowNextEventBasedSurvey()).toBe(false)
+    })
+
+    test('callSurveysAndEvaluateDisplayLogic should handle surveys correctly', () => {
+        mockPostHog.getActiveMatchingSurveys = jest.fn((callback) => callback([mockSurveys[0]]))
+
+        const handlePopoverSurveyMock = jest
+            .spyOn(surveyManager as any, 'handlePopoverSurvey')
+            .mockImplementation(() => {
+                console.log('Mock handlePopoverSurvey called')
+            })
+        const handleWidgetMock = jest.spyOn(surveyManager as any, 'handleWidget').mockImplementation(() => {})
+        const handleWidgetSelectorMock = jest
+            .spyOn(surveyManager as any, 'handleWidgetSelector')
+            .mockImplementation(() => {})
+        jest.spyOn(surveyManager as any, 'canShowNextEventBasedSurvey').mockReturnValue(true)
+
+        surveyManager.callSurveysAndEvaluateDisplayLogic()
+
+        expect(mockPostHog.getActiveMatchingSurveys).toHaveBeenCalled()
+        expect(handlePopoverSurveyMock).toHaveBeenCalledWith(mockSurveys[0])
+        expect(handleWidgetMock).not.toHaveBeenCalled()
+        expect(handleWidgetSelectorMock).not.toHaveBeenCalled()
+    })
+
+    test('handleWidget should render the widget correctly', () => {
+        const mockSurvey = mockSurveys[1]
+        const handleWidgetMock = jest.spyOn(surveyManager as any, 'handleWidget').mockImplementation(() => {})
+        surveyManager.getTestAPI().handleWidget(mockSurvey)
+        expect(handleWidgetMock).toHaveBeenCalledWith(mockSurvey)
+    })
+
+    test('handleWidgetSelector should set up the widget selector correctly', () => {
+        const mockSurvey = {
+            id: 'widgetSurvey2',
+            name: 'Widget Survey 2',
+            type: SurveyType.Widget,
+            appearance: { widgetType: 'selector', widgetSelector: '.widget-selector' },
+            start_date: '2021-01-01T00:00:00.000Z',
+            questions: [
+                {
+                    question: 'What would you like to see next?',
+                    type: SurveyQuestionType.Open,
+                },
+            ],
+            conditions: {},
+            end_date: null,
+            targeting_flag_key: null,
+        } as any
+        document.body.innerHTML = '<div class="widget-selector"></div>'
+        const handleWidgetSelectorMock = jest
+            .spyOn(surveyManager as any, 'handleWidgetSelector')
+            .mockImplementation(() => {})
+        surveyManager.getTestAPI().handleWidgetSelector(mockSurvey)
+        expect(handleWidgetSelectorMock).toHaveBeenCalledWith(mockSurvey)
```

Comment:
just a stronger assert 

```suggestion
        expect(handleWidgetSelectorMock).toHaveBeenCalledOnceWith(mockSurvey)
```